Rubberduck.Core/UI/Inspections/InspectionResultsViewModel.cs

```diff
@@ -232,6 +234,106 @@ public bool GroupByLocation
             }
         }
 
+        private List<CodeInspectionSeverity> _filterCriteria = new List<CodeInspectionSeverity>();
+
+        public bool FilterInspectionsByHint
+        {
+            get => _filterCriteria.Contains(CodeInspectionSeverity.Hint);
+            set
+            {
+                if (FilterInspectionsByHint == value) { return; }
+
+                if (value)
+                {
+                    _filterCriteria.Add(CodeInspectionSeverity.Hint);
+                }
+                else
+                {
+                    _filterCriteria.Remove(CodeInspectionSeverity.Hint);
+                }
+
+                FilterResults(_filterCriteria);
+                OnPropertyChanged();
+            }
+        }
+
+        public bool FilterInspectionsBySuggestion
+        {
+            get => _filterCriteria.Contains(CodeInspectionSeverity.Suggestion);
+            set
+            {
+                if (FilterInspectionsBySuggestion == value) { return; }
```

Comment:
All of these setters are virtually identical other than the `CodeInspectionSeverity`. I'd extract the common code to a single method.